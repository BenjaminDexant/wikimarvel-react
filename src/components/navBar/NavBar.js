import { Link } from "react-scroll";

import "./navBar.css";

const NavBar = () => {
	return (
		<ul className="list">
			<li className="element">
				<Link activeClass="active" to="home" spy={true} smooth={true}>
					Home
				</Link>
			</li>
			<li className="element">
				<Link to="searchCharacter" spy={true} smooth={true}>
					Search
				</Link>
			</li>
			<li className="element">
				<Link to="events" spy={true} smooth={true}>
					Events
				</Link>
			</li>
			<li className="element">
				<Link to="footer" spy={true} smooth={true}>
					Footer
				</Link>
			</li>
		</ul>
	);
};

export default NavBar;
