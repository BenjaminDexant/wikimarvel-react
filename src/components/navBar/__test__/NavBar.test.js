import React from "react";
import NavBar from "../NavBar";
import { fireEvent, render } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";

test("The navBar is made of an ul with four li elements Home Search Events and Footer", () => {
	const elem = renderer.create(<NavBar />).toJSON();
	expect(elem).toMatchSnapshot();
});

test("click to an element scroll to it", () => {
	const { getByTestId } = render(<NavBar />);
	const liElem = getByTestId("element");
	fireEvent.click(liElem, {
		target: {
			activeClass: "active",
		},
	});
	expect(liElem.activeClass).toBe("active");
});
