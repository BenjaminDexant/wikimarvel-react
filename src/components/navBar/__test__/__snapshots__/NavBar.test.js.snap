// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The navBar is made of an ul with four li elements Home Search Events and Footer 1`] = `
<ul
  className="list"
>
  <li
    className="element"
  >
    <a
      onClick={[Function]}
    >
      Home
    </a>
  </li>
  <li
    className="element"
    data-testid="element"
  >
    <a
      onClick={[Function]}
    >
      Search
    </a>
  </li>
  <li
    className="element"
  >
    <a
      onClick={[Function]}
    >
      Events
    </a>
  </li>
  <li
    className="element"
  >
    <a
      onClick={[Function]}
    >
      Footer
    </a>
  </li>
</ul>
`;
